### PR TITLE
Mozilla SSL Configuration Jul-2022

### LINKED ISSUE #125: Outdated Mozilla SSL configuration
As of today, modern profile do not support TLSv1.2 and intermediate does not support -TLSv1 -TLSv1.1. CipherSuite has also been updated.

### PR DESCRIPTION
**Pull request by staticdev**
Added latest recommended config for Apache and Nginx from:
* https://ssl-config.mozilla.org
* https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html

Closes #125

### CODE REVIEW

File: templates/apache2_nc.j2 [hunk 1]
```diff
@@ -54,6 +54,11 @@
   SSLCertificateChainFile {{ nextcloud_tls_cert_chain_file }}
 {% endif %}
 
+{% if (nextcloud_computed_major_version|int) > 13 %}
```
Comment (wiktor2200) [inline templates/apache2_nc.j2:57]: IDK if it's still a need to check Nextcloud version. Version should be higher than 13 in 100% and if it's not that's asking out trouble.
BTW versions < 22 are unsupported right now: https://nextcloud.com/changelog/

**Review by wiktor2200 — COMMENTED** [review]
I've checked https://ssl-config.mozilla.org for "modern" and "intermediate" and it looks good to me.

File: templates/nginx_nc.j2 [hunk 2]
```diff
@@ -36,6 +39,8 @@ server {
     ssl_session_timeout 1d;
```
[the remark this answers is not in the document]
Comment (staticdev) [reply templates/nginx_nc.j2:43]: Done.

File: templates/apache2_nc.j2 (hunk 1 above)
Comment (staticdev) [reply templates/apache2_nc.j2]: @wiktor2200 removed nextcloud < 13 code.

**Review by wiktor2200 — APPROVED** [review]
I've compared with Mozilla's SSL https://ssl-config.mozilla.org/ and it looks good to me now.

**Review by aalaesar — CHANGES_REQUESTED** [review]
I'm ok with removing support for nextcloud <14
but then https://github.com/aalaesar/install_nextcloud/blob/d49ea16389c756beb269ced2352ef368d20078a6/vars/main.yml#L13-35 become dead code.
Can you remove it please ? thanks 😄

**Comment by staticdev** [comment]
> I'm ok with removing support for nextcloud <14 but then [vars/main.yml line 15 to 35](https://github.com/aalaesar/install_nextcloud/blob/d49ea16389c756beb269ced2352ef368d20078a6/vars/main.yml#L13) become dead code. Can you remove it please ? thanks smile

Done.

**Review by aalaesar — APPROVED** [review]

**Pull request merged**